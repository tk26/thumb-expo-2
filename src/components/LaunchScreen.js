import React, { Component } from 'react';
import { View, Text, Image, Linking, Button } from 'react-native';

export default class LaunchScreen extends Component {
    render() {
        return (
            <View>
                <Image source={require('../../assets/thumb-horizontal-logo.png')} />
                <Text>Welcome to thumb</Text>
            
                <Button title="Login" onPress={() => this.props.navigation.navigate('LoginScreen')} />
                <Button title="Create Account" onPress={() => this.props.navigation.navigate('SignupStep1')}/>
                
                <Text style={{fontFamily: 'Roboto'}}>
                    By tapping Log In or Create Account, I agree to thumb's&nbsp;
                    <Text style={{ color: 'blue' }} onPress={() => Linking.openURL('https://www.google.com')}>
                        Terms of Service
                    </Text>
                    ,&nbsp;
                    <Text style={{ color: 'blue' }} onPress={() => Linking.openURL('https://www.google.com')}>
                        Privacy Policy
                    </Text>
                    , and&nbsp;
                    <Text style={{ color: 'blue' }} onPress={() => Linking.openURL('https://www.google.com')}>
                        Non-discrimination Policy
                    </Text>
                    .
                </Text>
            </View>
        );
    }
}