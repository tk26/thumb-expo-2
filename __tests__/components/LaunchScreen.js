import React from 'react';
import renderer from 'react-test-renderer';
import LaunchScreen from '../../src/components/LaunchScreen';

test('renders correctly', () => {
  const tree = renderer.create(<LaunchScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
