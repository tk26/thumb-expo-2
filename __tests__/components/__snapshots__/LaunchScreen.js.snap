// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "padding": 10,
    }
  }
>
  <View
    style={
      Object {
        "elevation": 1,
        "marginLeft": 5,
        "marginRight": 5,
        "marginTop": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 50,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 5,
            "position": "relative",
          },
          undefined,
        ]
      }
    >
      <Image
        source={1}
        style={
          Object {
            "height": 120,
            "resizeMode": "contain",
            "width": 120,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 5,
            "position": "relative",
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#212121",
            "fontFamily": "HelveticaNeue-Bold",
            "fontSize": 44,
            "letterSpacing": 0,
            "lineHeight": 43,
          }
        }
      >
        Welcome to thumb
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": 30,
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 5,
            "position": "relative",
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#fff",
            "maxWidth": 150,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "backgroundColor": "#98FB98",
              "bottom": 0,
              "height": 8,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 18,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "#424242",
              "fontFamily": "Helvetica Neue",
              "fontSize": 16,
              "fontWeight": "600",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        >
          log in
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "width": 30,
          }
        }
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#fff",
            "maxWidth": 150,
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Object {
              "backgroundColor": "#98FB98",
              "bottom": 0,
              "height": 8,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 18,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "alignSelf": "center",
              "color": "#424242",
              "fontFamily": "Helvetica Neue",
              "fontSize": 16,
              "fontWeight": "600",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            }
          }
        >
          create account
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "padding": 5,
            "position": "relative",
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#424242",
            "fontFamily": "Helvetica Neue",
            "fontSize": 13,
            "letterSpacing": 0,
            "lineHeight": 18,
          }
        }
      >
        By tapping "log in" or "create account", I agree to thumb's 
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          onPress={[Function]}
          style={
            Object {
              "color": "#424242",
              "fontFamily": "Helvetica Neue",
              "fontSize": 12,
              "letterSpacing": 1.2,
              "lineHeight": 16,
              "textDecorationColor": "#424242",
              "textDecorationLine": "underline",
            }
          }
        >
          TERMS OF SERVICE
        </Text>
         and 
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          onPress={[Function]}
          style={
            Object {
              "color": "#424242",
              "fontFamily": "Helvetica Neue",
              "fontSize": 12,
              "letterSpacing": 1.2,
              "lineHeight": 16,
              "textDecorationColor": "#424242",
              "textDecorationLine": "underline",
            }
          }
        >
          PRIVACY POLICY
        </Text>
        .
      </Text>
    </View>
  </View>
</View>
`;
